src/client/datascience/interactive-ipynb/nativeEditor.ts

```diff
@@ -233,6 +236,58 @@ export class NativeEditor extends InteractiveBase implements INotebookEditor {
         this.previouslyNotTrusted = !this._model.isTrusted;
     }
 
+    @captureTelemetry(Telemetry.SyncAllCells)
+    public async syncAllCells(): Promise<void> {
+        const timer = new StopWatch();
+        // Ask the UI for all of our all cells
+        this.waitingForMessageResponse = {
+            message: InteractiveWindowMessages.ReturnAllCellCode,
+            result: createDeferred<string[]>()
+        };
+        await this.postMessage(InteractiveWindowMessages.GetAllCellCode);
+        const code = await this.waitingForMessageResponse.result.promise;
```

Comment:
This method will never complete if we have another request for `syncAllCell`.
For the second execution of `syncAllCells`, the `result` points to another deferred.
Hence the first request will never complete.

Thus when saving, only the last save wins, I guess its ok. Last one wins.
If this is by design then fine, else concerned that if we have auto save, then it will never save until the user stops typing or doing something (because as the user types, we get more of these getting into the pipeline and the subsequent saves result in new promises, meaning only last one wins).

Reply:
Yes good point. I'll make it a list and pass in an id.